examples/set-a-custom-referrer/index.js

```diff
@@ -0,0 +1,31 @@
+import { RequestLogger, RequestHook } from "testcafe";
+
+fixture `Set a Custom Referer`
+    .page`http://example.com/`;
+
+export class MyRequestHook extends RequestHook {
+    constructor(requestFilterRules, responseEventConfigureOpts) {
+        super(requestFilterRules, responseEventConfigureOpts);
+    }
+    async onRequest(event) {
+        event.requestOptions.headers["Referer"] =
+            "http://my-modified-referer.com";
+    }
+    async onResponse(event) {
+
+    }
+}
+
+const hook = new MyRequestHook();
+const logger = RequestLogger("http://example.com/", {
+    logRequestHeaders: true,
+});
+
+test.requestHooks(hook, logger)("Check the Referer Value", async t => {
```

Comment:
Formatting

```js
test
    .requestHooks([hook, logger])
    ('Check the Referer Value', async t => {
           await t
                .navigateTo("https://devexpress.github.io/testcafe/example/")
                .expect(logger.contains(r => r.request.url === 'https://devexpress.github.io/testcafe/example/').ok() // wait for navigation completed
                .expect(logger.requests[0].request.headers['referrer']).eql('http://my-modified-referer.com');
    });
```

Reply:
the following assertion always fails because the logger is not set to track the '/testcafe/example' URL.

> .expect(logger.contains(r => r.request.url === 'https://devexpress.github.io/testcafe/example/').ok()

 I've modified the logger above to track all urls:

> const logger = RequestLogger(**null**, {
    logRequestHeaders: true,
});